Test/ViewSupportSpec.hs

```diff
@@ -0,0 +1,111 @@
+{-|
+Module: Test.ViewSupportSpec
+
+Tests for view support functions.
+-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+module Test.ViewSupportSpec where
+import qualified Prelude
+import ClassyPrelude
+import Test.Hspec
+import IHP.Test.Mocking hiding (application)
+import IHP.Prelude
+import IHP.QueryBuilder
+import IHP.Environment
+import IHP.FrameworkConfig
+import IHP.HaskellSupport
+import IHP.RouterSupport hiding (get)
+import IHP.FrameworkConfig
+import IHP.Job.Types
+import IHP.Controller.RequestContext hiding (request)
+import IHP.ViewPrelude
+import IHP.ControllerPrelude hiding (get, request)
+import qualified IHP.Server as Server
+import Data.Attoparsec.ByteString.Char8 (string, Parser, (<?>), parseOnly, take, endOfInput, choice, takeTill, takeByteString)
+import Network.Wai
+import Network.Wai.Test
+import Network.HTTP.Types
+import qualified IHP.ErrorController as ErrorController
+import Data.String.Conversions
+import Data.Text as Text
+import Unsafe.Coerce
+import IHP.ApplicationContext
+
+data WebApplication = WebApplication deriving (Eq, Show, Data)
+
+data TestController
+    = TestAction
+    | TestWithParamAction { param :: Text }
+  deriving (Eq, Show, Data)
+
+data AnotherTestController
+    = AnotherTestAction
+  deriving (Eq, Show, Data)
+
+instance Controller TestController where
+    action TestAction = do
+        renderPlain "TestAction"
+    action TestWithParamAction { .. } = do
+        let output = [plain|
+            isActiveAction #{param}: #{isActiveAction $ TestWithParamAction param}
+            isActiveAction bar: #{isActiveAction $ TestWithParamAction "bar"}
+
+            isActivePath #{param}: #{isActivePath $ "/test/TestWithParam?param=" <> param}
+            isActivePath bar: #{isActivePath ("/test/TestWithParam?param=bar" :: Text)}
+        |]
```

Comment:
@mpscholten added tests so we have more confidence over the change! :beers: 

When I try to add

`isActiveController TestController: #{isActiveController @TestController}`

I get:

![image](https://github.com/digitallyinduced/ihp/assets/125707/5fd65442-f59b-4470-a00b-20db8f6ac368)

Reply:
The function can only be called from a view. To fix the test you can add a view data structure and render it to HSX

Reply:
> The function can only be called from a view

What indicates it? Is it the use of `fromFrozenContext`?

Reply:
Yep exactly. The `renderHtml` does call `Context.freeze` before rendering the html https://github.com/digitallyinduced/ihp/blob/master/IHP/Controller/Render.hs#L60

Reply:
Some explanation behind this is here https://github.com/digitallyinduced/ihp/blob/master/IHP/Controller/Context.hs#L14

Reply:
@mpscholten I've tried to add a `View` in https://github.com/digitallyinduced/ihp/pull/1725/commits/f790af3233b3cee01c1dfc6cc6e91a21b3a1318f  but I'm getting a 500 error, but I don't know how to get the error message. Trying to print the output of the body ```"" `shouldBe` body``` showed an empty string

![image](https://github.com/digitallyinduced/ihp/assets/125707/39fdf244-0a89-4f01-a3b4-fa08e25cf50e)

Reply:
@mpscholten If it's not an easy fix, I can remove the `isControllerActive` test, in favor of getting this in.

Reply:
@mpscholten ^^

Reply:
Sorry, forgot to comment here, pushed a fix to https://github.com/amitaibu/ihp/pull/4/files Once merged the tests should be green 👍

Reply:
Thanks. @mpscholten can you give a short explanation on how you knew what this fix is?

Reply:
I dumped the response body of the the 500 Internal Server Error response. It showed that the error was happening here https://github.com/digitallyinduced/ihp/blob/master/IHP/Controller/Session.hs#L162 That error should never happen. I was first confused why the session was even used, because no code in the tests is doing anything with the session. I placed many `error` calls in my local IHP codebase to trace it back. Turns out that [`renderHtml` calls `initFlashMessages` which then accesses the session](https://github.com/digitallyinduced/ihp/blob/master/IHP/Controller/Render.hs#L58). 

Then I looked thorugh the test code and checked what could be wrong with the session setup. I figured out that we never called the session middleware. After that it was an easy fix, just wiring up the middleware so that the session is available 👍

Reply:
> I dumped the response body of the the 500 Internal Server Error response

How did you do it?

Reply:
Changed the

```haskell
assertTextExists body response = do
    response.simpleStatus `shouldBe` status200
```

function to this:

```haskell
assertTextExists body response = do
    error (show response)
    response.simpleStatus `shouldBe` status200
```